### DISCUSSION TITLE
More Windows 11 fitting, intuitive Properties window

### DISCUSSION THREAD

**Discussion by dongle-the-gadget** [valinet/ExplorerPatcher#463] [Ideas]
The Properties window doesn't quite fit the Windows 11 UI designs and also is relatively unintuitive (i.e. I have to click on text to get options, should use a drop down box instead).
I would suggest using PowerToys' Settings UI would be a good start.

**Comment by valinet** [comment]
The Properties UI is an afterthought, it was never intended to be shipped in the first place, I created it eventually because some people simply insisted a lot; it frankly represents extra, non-essential functionality.

The current implementation is designed so that the UI tree is very easy to maintain and extend with new options (the interface is built programmatically from a file describing it at very high level), it is very lightweight and uses very few resources, pulls in no clutter or additional dependencies compared to pretty much any other UI framework including WinUI (like restricting the implementation to a certain language, as WinUI does, which mandates me not to write in C for some reason) and it was relatively quick to implement. It allows me to satisfy this need in a good enough manner while being able to concentrate on the things that matter, like bringing actual functionality. It's an original development, built from scratch, and in line with the focus of this project, where the idea is to use mostly in-house stuff rather then burden the project with tons of dependencies. It's functional and that is what matters primarily for a thing of its kind.

Bells and whistles like effects when hovering items / for drop downs are not implemented because they'd add nothing functionally, hence I do not have time for that, not to mention they introduce the possibility for bugs. I think it's plenty intuitive anyway if one dares to spend more than 5 seconds trying to work with it. Of course, contributions can be made in the area as well, but taking into consideration the general idea behind the original implementation.

Personally, I do not have any plans to work in this area. The investment does not make sense at all to me, I am not interested in devoting huge amounts of time learning yet another UI framework, I am more interested in exploring other areas regarding this project. I am not a graphical designer. Reshaping the interface using any other framework while keeping the current flexibility of the backend is a task I simply do not have the time or mood for. I don't want to spend all the free time I have working on this tool, and it more than does its job well enough for now and it takes a lot of time anyway even currently.

With these being said, if you have a definite plan on how to implement this, be my guest, we can talk about it. But personally, I won't be doing work on this direction.

Thanks

**Reply by dongle-the-gadget** [reply]
For the WinUI problem, we could somewhat get around that by using XAML Islands, but XAML Islands as its own entity is already a pain to work with anyway.

**Reply by BraINstinct0** [reply]
IIRC this whole setting thing was created to make customisations a tad easier(I think I was the one who requested it; not sureif it was already planned though).
If the UI overhauling effort make development of ExplorerPatcher functionalities slower, I'd just be happy with the current UI. After all, it's mostly scripts&registry edits; given enough time someone with extensive C++/GUI programming (WPF/WinForms/XAML) might recreate it, but that's not me...

**Reply by dongle-the-gadget** [reply]
I mean if there are some documentation around the registry keys and stuff, I could try to make one.
(for the record I am not skilled at C)

**Comment by Anixx** [comment]
Well, I agree. The drop-down boxes do not look like drop-down boxes, and buttons (such as update button) look like simple text.

Sometimes the entries look like plain text even though they are interactive.
Also, the checkboxes are located to the left of the description, while the drop-down boxes are located to the right. Maybe move the checkboxes to the right as well?

And, possibly, separate the entries with thick dots?

For instance, here:
![изображение](https://user-images.githubusercontent.com/1271736/143243595-75edac8b-1cfe-422c-87e8-9725fdf9710e.png)

* The first entry is a drop-down
* Then goes 3 links that do not affect anything, but with an unfamiliar arrow character, so some people may think it is an option switch.
* Then go 5 lines, of which 4 are drop-downs but separated only by space. The first of them ends with a colon, which makes impression it is similar to the other lines, a drop-down with currently empty value.

![изображение](https://user-images.githubusercontent.com/1271736/143244076-c9b1cb92-5367-40b6-88a7-46cc3a6a6a1d.png)

Here the both description of an option and its value are very long, and not separated by style or other signs, so making impression it is single text (notice also the repeated twice colons).

![изображение](https://user-images.githubusercontent.com/1271736/143244313-7bf9e966-a525-424d-a743-dee085afe446.png)

Here, the checkboxes are on the left, but the drop-down is on the right, making impression that the last line is a footnote (the real footnote is located in the bottom of the page).

![изображение](https://user-images.githubusercontent.com/1271736/143244703-021141ef-ea0f-4c7e-900a-35a99d648938.png)

Here again, the first line looks like a descriptive phrase, given that the drop-down value is bigger than the label description and of the same style.

The update buttons look like hyperlinks. But unlike hyperlinks, there is no finger cursor over them, so they make impression as simply bold text. Cliking on them also does not provide feedback. Only after a few seconds after clicking on the second one you realize it worked and was a button (as the Explorer restarts). Normally after clicking on underlined text you would expect a page opened in a browser. After clicking on the first one, nothing happens at all for me, as the notifications are disabled.

Maybe it could be considered to use some more standard Windows controls?... Or at least, make it obvious, where is the text and where are the choice options, by using different style or color?
I also suggest to move the checkboxes to the ends of each lines, and start all option line (checkbox or drop-down or link) with a thick dot so to separate them.

**Reply by valinet** [reply]
As I said, the UI remains like this. I am negatively surprised it generates so much debate from some of you considering it is used once to configure the app and then left alone in peace.

I already said I don't have time to nit pick it at the level of your posts. I also have to actually use the computer. Sometimes I wonder if it's the same case with you. It looks to me that you spend way more time nit picking stuff in the interface than you do actually using it, which seems counterproductive to me. I understand, the UI maybe is not the best, okay? I get that. But it works, it gets the job done, so it's at an acceptable level in my opinion. Not to mention some of the remarks are questionable at best, coming down to personal preference... For example:

> with an unfamiliar arrow character, so some people may think it is an option switch.

"🡕" is a well-known symbol for external links, except in 1995 probably, indeed. Otherwise, it's understandable from context what it does just fine.

> (notice also the repeated twice colons).

What????

Want something else? The source is available in the repo; I keep seeing tons of posts from you but no actual plan of action that you have regarding what you describe in them. Contributions I mean. I can't keep up with these many demands, as I said, I have other stuff to do as well. Furthermore, the direction I envision for this project may at some point diverge from the direction you want it taken, so at that point, I'd suggest forking it and implementing whatever stuff you personally see worth implementing there.

Unless there are some contributions, I don't see the purpose of posts where every small inconsistency is nit picked. Bugs and punctual requests are one thing, but how a UI looks is something else, an endless debate imo. Why don't you tell the whole story though? If I remake the interface in WinUI, for example, as suggested by the OP, you will come again crying out loud how the UI is not looking well with classic theme etc etc. It's always a matter of taste and it's subjective. I did it this way. Don't like it? Fine, there's always room for creating an alternative implementation. If it gets to feature parity with the current one, provides meaningful improvements and you would be so kind as to contribute it to the upstream, I'd gladly accept it, at least as an alternative if not as the default. But without that, I already told you what my position regarding this is.

Thanks

**Comment by dongle-the-gadget** [comment]
There are also some weird oddities within this like
![Content of dialog shifted to top left corner](https://user-images.githubusercontent.com/29563098/143264738-47340b62-fcdc-4bab-ad58-745e7bec0fba.png)

**Reply by valinet** [reply]
How do you get it into this condition?

**Reply by dongle-the-gadget** [reply]
Just by like randomly clicking around the empty spaces.

**Reply by valinet** [reply]
Thanks. Very helpful to reproduce. /s Can you determine a pattern and then tell me what is it? Or at least a hint…?

**Reply by valinet** [reply]
Yeah, clicked randomly a couple of times now, cannot reproduce. Idk, the only thing that could be different is the title of the window, which is localized, mine is US English, so "Properties". Again, the most helpful of the bug reports. And no, it's not a feature, or an oddity. It's a problem, if it really happens naturally, without inducing it in some way.

**Reply by dongle-the-gadget** [reply]
> Yeah, clicked randomly a couple of times now, cannot reproduce. Idk, the only thing that could be different is the title of the window, which is localized, mine is US English, so "Properties". Again, the most helpful of the bug reports. And no, it's not a feature, or an oddity. It's a problem, if it really happens naturally, without inducing it in some way.

Strange.

**Reply by BraINstinct0** [reply]
> Just by like randomly clicking around the empty spaces.

Maybe post a short vid/img of where you clicked?
